lib/Parse/ParseExpr.cpp

```diff
@@ -538,6 +539,8 @@ ParserResult<Expr> Parser::parseExprKeyPath() {
   }
 
   if (startsWithSymbol(Tok, '.')) {
+    llvm::SaveAndRestore<Expr*> S(SwiftKeyPathRoot, rootResult.getPtrOrNull());
```

Comment:
If we are not using SaveAndRestore, we are not able to use this part in parseExprPostfixSuffix.